Unit tests aren't re-run if they fail in next make

Say you run 

    make

...and the unit tests fail.

Then next time you run

    make

the unit tests won't be re-run and the build will complete successfully.

This is probably because the make rule uses file test_results.log alone as a target in order to determine if it needs to execute.

After I make clean:

1. On first make nidmm unit tests fail, build stops.
2. On second make nimodinst unit tests fail, build stops.
3. On third make, "success"!

```
Kirsch0:nimi-python kirsch$ make
make[1]: Nothing to be done for `MKDIR'.
Creating nidmm enums.py
Creating nidmm library.py
Creating nidmm session.py
Creating nidmm errors.py
Creating nidmm ctypes_library.py
Creating nidmm mock_helper.py
Creating nidmm __init__.py
Creating nidmm ctypes_types.py
Creating nidmm python_types.py
Creating nidmm session.rst
Creating nidmm enums.rst
Creating nidmm attributes.rst
Creating nidmm test_session.py
Creating nidmm setup.py
Updating generated files for nidmm
Creating nidmm README.rst
Running unit tests for nidmm
make[1]: *** [/Users/kirsch/Developer/nimi-python/bin/nidmm/log/test_results.log] Error 1
make: *** [nidmm] Error 2
Kirsch0:nimi-python kirsch$ make
make[1]: Nothing to be done for `MKDIR'.
Updating generated files for nidmm
Creating Wheel distribution for nidmm
zip_safe flag not set; analyzing archive contents...
Creating Source distribution for nidmm
Creating nimodinst library.py
Creating nimodinst session.py
Creating nimodinst errors.py
Creating nimodinst ctypes_library.py
Creating nimodinst mock_helper.py
Creating nimodinst __init__.py
Creating nimodinst ctypes_types.py
Creating nimodinst python_types.py
Creating nimodinst session.rst
Creating nimodinst attributes.rst
Creating nimodinst test_modinst.py
Creating nimodinst setup.py
Updating generated files for nimodinst
Creating nimodinst README.rst
Running unit tests for nimodinst
make[1]: *** [/Users/kirsch/Developer/nimi-python/bin/nimodinst/log/test_results.log] Error 2
make: *** [nimodinst] Error 2
Kirsch0:nimi-python kirsch$ make
make[1]: Nothing to be done for `MKDIR'.
Updating generated files for nidmm
Creating Wheel distribution for nidmm
Creating Source distribution for nidmm
make[1]: Nothing to be done for `MKDIR'.
Updating generated files for nimodinst
Creating Wheel distribution for nimodinst
zip_safe flag not set; analyzing archive contents...
Creating Source distribution for nimodinst
Kirsch0:nimi-python kirsch$ 

```

--- a/build/rules.mak
+++ b/build/rules.mak
@@ -54,7 +54,7 @@ module: $(MODULE_FILES)
 $(UNIT_TEST_FILES): $(MODULE_FILES) $(RST_FILES)
 unit_tests: $(UNIT_TEST_FILES)
 
-$(LOG_DIR)/test_results.log:
+$(LOG_DIR)/test_results.log: unit_tests
 	@echo Running unit tests for $(DRIVER)
 	$(_hide_cmds)$(call log_command,cd $(OUTPUT_DIR) && python3 -m pytest -s $(LOG_OUTPUT) $(LOG_DIR)/test_results.log)
 

